python/sglang/srt/models/deepseek_v2.py

```diff
@@ -1342,7 +1347,33 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
             ("gate_up_proj", "gate_proj", 0),
             ("gate_up_proj", "up_proj", 1),
         ]
-
+        if self.n_share_fusion_experts != 0:
+            weights_list = list(weights)
+            weights_dict = {k: v for (k, v) in weights_list}
+            suffix_list = [
+                'down_proj.weight', 'down_proj.weight_scale_inv',
+                'gate_proj.weight', 'gate_proj.weight_scale_inv',
+                'up_proj.weight', 'up_proj.weight_scale_inv'
+            ]
+            current_device = torch.cuda.current_device()
```

Comment:
nit: it seems we need to consider platforms other than CUDA, so I saw code using get_device_module etc